extensions/wikia/AdEngine/js/AdEngineRunner.js

```diff
@@ -0,0 +1,107 @@
+/*global define, require*/
+define('ext.wikia.adEngine.adEngineRunner', [
+	'ext.wikia.adEngine.adEngine',
+	'ext.wikia.adEngine.adTracker',
+	'wikia.log',
+	'wikia.window',
+	require.optional('ext.wikia.adEngine.lookup.amazonMatch'),
+	require.optional('ext.wikia.adEngine.lookup.rubiconFastlane')
+], function (adEngine, adTracker, log, win, amazonMatch, rubiconFastlane) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.adEngineRunner',
+		supportedBidders = [amazonMatch, rubiconFastlane],
+		timeout = 2000;
+
+	/**
+	 * Delay running AdEngine by bidder responses or by configured timeout
+	 *
+	 * @param {function} runAdEngine
+	 */
+	function delayRun(runAdEngine) {
+		var enabledBidders = [],
+			biddersQueue = [];
+
+		/**
+		 * Mark bidder as responded and trigger run if all bidders already responded
+		 *
+		 * @param {string} name
+		 */
+		function markBidder(name) {
+			log(name + ' responded', 'debug', logGroup);
+			if (biddersQueue.indexOf(name) === -1) {
+				biddersQueue.push(name);
+			}
+			if (biddersQueue.length === enabledBidders.length) {
+				log('All bidders responded', 'info', logGroup);
+				runAdEngine();
+			}
+		}
+
+		/**
+		 * Add bidder listener to mark bidder on response
+		 */
+		function registerBidders() {
+			log(['Register bidders', enabledBidders], 'debug', logGroup);
```

Comment:
After QA notes: this does not give us anything. Would be better just to pass bidder module's name to the log ;)
![register-bidders-log](https://cloud.githubusercontent.com/assets/1662451/12587610/6a74a020-c455-11e5-90c9-f28d50804be3.png)